DSharpPlus.Core/DiscordClient.cs

```diff
@@ -0,0 +1,124 @@
+                .AddMemoryCache();
+
+            Services = serviceCollection.BuildServiceProvider();
+
+            #region set up policies, retrying etc. this should be moved to a separate method.
+            PollyRatelimitingPolicy ratelimiter = new();
+
+            // these values are arbitrary and should be configurable. they specify how much time should elapse between retries,
+            // and how often we should retry. the algorithm creates an exponentially increasing list of delays, with the first one
+            // being based on 0.5 seconds here.
+            IEnumerable<TimeSpan> retryDelay = Backoff.DecorrelatedJitterBackoffV2(TimeSpan.FromSeconds(0.5), 10);
+
+            ServiceCollection.AddHttpClient<RestClient>()
+                .ConfigureHttpClient((services, httpClient) =>
+                {
+                    httpClient.BaseAddress = new("https://discord.com/api/v9"); // or whatever, ideally have this in a constant
+                    httpClient.DefaultRequestHeaders.UserAgent.Add(new("DSharpPlus", "v5")); // again, constants
+                })
+                // register our ratelimiter
+                .AddTransientHttpErrorPolicy(policy => policy.WaitAndRetryAsync(retryDelay).WrapAsync(ratelimiter))
+
+                // register our retrying
+                .AddPolicyHandler
+                (
+                    // only retry 429 in addition to the already specified 5xx and 408
+                    Policy.HandleResult<HttpResponseMessage>(result => result.StatusCode == HttpStatusCode.TooManyRequests)
+                        .WaitAndRetryAsync(retryCount: 1, // retry once. another arbitrary number.
+                        sleepDurationProvider: (_, response, _) =>
+                        {
+                            HttpResponseMessage message = response.Result;
+
+                            if (message.Headers.GetValues("X-RateLimit-Scope").SingleOrDefault() == "shared")
+                            {
+                                throw new HttpRequestException("Shared ratelimit hit, not retrying request."); // this should be our own exception type
+                            }
+
+                            else if (message == default)
+                            {
+                                return oneSecond;
+                            }
+#pragma warning disable IDE0046 // shut up about ternaries
+                            else if (message.Headers.RetryAfter == null || message.Headers.RetryAfter.Delta == null)
+#pragma warning restore IDE0046
+                            {
+                                return oneSecond;
+                            }
+                            else 
+                            { 
+                                return message.Headers.RetryAfter.Delta.Value; 
+                            }
+                        },
+                        onRetryAsync: (_, _, _, _) => Task.CompletedTask)
+                );
```

Comment:
This should be a proper policy class imo.

Reply:
it could, sure. one issue with that is the fact i'd like to have these magic numbers configurable, which wouldn't really help readability if this is far away from the configuration